### ISSUE TITLE
`ConditionalOnProperty` target a property that has no metadata

### ISSUE THREAD

**Issue by snicoll** [spring-cloud/spring-cloud-commons#407]
Several auto-configurations use a condition on a property that has no metadata. As a result, trying to use it to customize the auto-configuration lead to a warning in an IDE.

The report detected the following:

```
org.springframework.cloud.autoconfigure.RefreshAutoConfiguration [spring.cloud.refresh.enabled]
org.springframework.cloud.commons.httpclient.HttpClientConfiguration.ApacheHttpClientConfiguration [spring.cloud.httpclientfactories.apache.enabled]
org.springframework.cloud.commons.httpclient.HttpClientConfiguration.OkHttpClientConfiguration [spring.cloud.httpclientfactories.ok.enabled]
org.springframework.cloud.commons.util.UtilAutoConfiguration [spring.cloud.util.enabled]
org.springframework.cloud.client.CommonsClientAutoConfiguration.ActuatorConfiguration [spring.cloud.features.enabled]
org.springframework.cloud.client.CommonsClientAutoConfiguration.DiscoveryLoadBalancerConfiguration [spring.cloud.discovery.enabled]
org.springframework.cloud.client.CommonsClientAutoConfiguration.DiscoveryLoadBalancerConfiguration#discoveryCompositeHealthIndicator [spring.cloud.discovery.client.composite-indicator.enabled]

```

**Comment by ryanjbaxter** [comment]
I should just create a PR but oh well ;)

@snicoll does this seem to be the correct format for this file?
```
{
  "hints": [],
  "groups": [],
  "properties": [
    {
      "sourceType": "org.springframework.cloud.autoconfigure.RefreshAutoConfiguration",
      "defaultValue": "true",
      "name": "spring.cloud.refresh.enabled",
      "description": "Enables autoconfiguration for the refresh scope and associated features.",
      "type": "java.lang.Boolean"
    },
    {
      "sourceType": "org.springframework.cloud.commons.httpclient.HttpClientConfiguration.ApacheHttpClientConfiguration",
      "defaultValue": true,
      "name": "spring.cloud.httpclientfactories.apache.enabled",
      "description": "Enables creation of Apache Http Client factory beans.",
      "type": "java.lang.Boolean"
    },
    {
      "sourceType": "org.springframework.cloud.commons.httpclient.HttpClientConfiguration.OkHttpClientConfiguration",
      "defaultValue": "true",
      "name": "spring.cloud.httpclientfactories.ok.enabled",
      "description": "Enables creation of OK Http Client factory beans.",
      "type": "java.lang.Boolean"
    },
    {
      "sourceType": "org.springframework.cloud.commons.util.UtilAutoConfiguration",
      "defaultValue": true,
      "name": "spring.cloud.util.enabled",
      "description": "Enables creation of Spring Cloud utility beans.",
      "type": "java.lang.Boolean"
    },
    {
      "sourceType": "org.springframework.cloud.client.CommonsClientAutoConfiguration.ActuatorConfiguration",
      "defaultValue": true,
      "name": "spring.cloud.features.enabled",
      "description": "Enables the features endpoint.",
      "type": "java.lang.Boolean"
    },
    {
      "sourceType": "org.springframework.cloud.client.CommonsClientAutoConfiguration.DiscoveryLoadBalancerConfiguration",
      "defaultValue": true,
      "name": "spring.cloud.discovery.enabled",
      "description": "Enables discovery client health indicators.",
      "type": "java.lang.Boolean"
    },
    {
      "sourceType": "org.springframework.cloud.client.CommonsClientAutoConfiguration.DiscoveryLoadBalancerConfiguration",
      "defaultValue": true,
      "name": "spring.cloud.discovery.client.composite-indicator.enabled",
      "description": "Enables discovery client composite health indicator.",
      "type": "java.lang.Boolean"
    }
  ]
}
```

**Comment by snicoll** [comment]
Thanks for the feedback. a PR would be nice to make sure we're talking about the exact same thing (in particular the file location).

The `hints` and `groups` arrays are optional so you can remove them. This looks good, running the build and make sure those keys resolve is a good way to make sure the file is at the proper location.

**Comment by ryanjbaxter** [comment]
Closed via  99a3b6e
